app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/HomeViewModel.kt

```diff
@@ -52,8 +51,21 @@ class HomeViewModel @Inject constructor(
                 is EateryApiResponse.Pending -> EateryApiResponse.Pending
                 is EateryApiResponse.Success -> {
                     EateryApiResponse.Success(
-                        apiResponse.data.filter {
-                            passesFilter(it, filters, favorites, null)
+                        apiResponse.data.filter { eatery ->
+                            filters.all { filter ->
+                                when (filter) {
+                                    is Filter.FromEatery -> filter.passesFilter(eatery)
+                                    is Filter.RequiresFavoriteEateries -> filter.passesFilter(
+                                        eatery,
+                                        favorites
+                                    )
+
+                                    is Filter.CustomFilter -> when (filter) {
+                                        Filter.CustomFilter.Selected,
+                                        Filter.CustomFilter.Today -> true
+                                    }
+                                }
+                            }
```

Comment:
Hmm okay so this is definitely a big improvement on the filtering we had before (removing `passesFilter` function ENTIRELY is massive!) but I do think there's one final improvement we can make.

Note how the pattern matching on `is Filter.FromEatery` and `is Filter.RequiresFavoriteEateries` is a bit redundant in that it will ALWAYS be those results every time: `filter.passesFilter(eatery)` and `filter.passesFilter(
                                        eatery,
                                        favorites
                                    )`. In a sense, what we had before was nice in that we could just call `passesFilter(it, filters, favorites, null)` and it would be done.

I propose we write a function on `Filter` that does this pattern matching for us--so it will take in the eatery, favorites, (and trivially itself--`this`). This would do the `FromEatery` and `RequiresFavoriteEateries` redundant code away so that we don't have to keep pattern matching on it.

But then finally, since we do need to still care about `CustomFilter`, it can take in an anonymous function that, if it's custom filter, will do the custom logic for filtering that's passed in. 

This just removes some redundancy! LMK what you think.